### PR TITLE
Remaking the rpi support PR

### PR DESCRIPTION
**Pull request by bholland**
## Checklist

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] I used the same coding style as the surrounding code 
- [ ] I have tested the proposed changes
- [ ] I extended the documentation, if necessary
- [ ] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

**Comment by bholland** [comment]
This incorporates the diff from the other PR. This does work as well. I have a smooth 30.214 FPS at 720P on the rpi4. I think the other trick is to use SLD2. I don't know how to make this work but there is a ton of text and messages about finding OpenGL and GLX and more. For the PI, I would assume that those are irrelevant. Anyway, this does work. Let me know if you have other issues.

**Comment by lynxlynxlynx** [comment]
Are you saying it's not finding SDL2 for you?

File: CMakeLists.txt [hunk 1]
```diff
@@ -300,6 +313,26 @@ IF((NOT OPENGL_BACKEND STREQUAL "None") OR USE_OPENGL)
 	IF(SDL_BACKEND STREQUAL "SDL")
 		MESSAGE(FATAL_ERROR "SDL2 is required for the OpenGL driver!")
 	ENDIF()
+	IF(RPI)
+		MESSAGE(STATUS "RPI_GLES_PATH: ${RPI_GLES_PATH}")
+		MESSAGE(STATUS "RPI: ${RPI}")
+		MESSAGE(STATUS "Setting _OPENGL_INCLUDE_PATH: ${RPI_GLES_PATH}/include")
+		MESSAGE(STATUS "Setting _OPENGL_LIB_PATH: ${RPI_GLES_PATH}/libs")                                                                                                         
+                SET(_OPENGL_INCLUDE_PATH ${RPI_GLES_PATH}/include)
```
Comment (lynxlynxlynx) [inline CMakeLists.txt:321]:
```suggestion
	SET(_OPENGL_INCLUDE_PATH ${RPI_GLES_PATH}/include)
```

**Comment by bholland** [comment]
It finds SDL2 fine. It is just very hard to figure out what is going on
with OpenGL. It finds OpenGL, GLX, GLES, and other experimental libraries
that got installed by default. This code overrides all of that for rpi but
it doesn't seem to link to the GLES libraries in /opt/vc. It properly uses
the headers though and SDL2 links to the libraries properly.

Basically, whatever is happening works because I get 30FPS rather than 5. I
dont know if it is a mix between how GemRB links with my sdl2, if it is the
GLES libraries linking into GemRB, or if it is SDL2 linking and GemRB using
my SDL2.

I am not even sure how to test it out but I can say that this does work as
long as I complied my own SDL2 library because the raspbian package does
not link correctly.

**Comment by lynxlynxlynx** [comment]
Do you get different results if you ifdef the findopengl call (besides fewer messages)?

**Comment by bholland** [comment]
I can test it out tonight. Can you send me a diff?

**Comment by lynxlynxlynx** [comment]
It's right after the new stuff for EGL, so just IF(RPI) around it.

**Comment by lynxlynxlynx** [comment]
Any news, have you managed to try it?

File: CMakeLists.txt [hunk 2]
```diff
@@ -300,6 +313,26 @@ IF((NOT OPENGL_BACKEND STREQUAL "None") OR USE_OPENGL)
 	IF(SDL_BACKEND STREQUAL "SDL")
 		MESSAGE(FATAL_ERROR "SDL2 is required for the OpenGL driver!")
 	ENDIF()
+	IF(RPI)
+		MESSAGE(STATUS "RPI_GLES_PATH: ${RPI_GLES_PATH}")
+		MESSAGE(STATUS "RPI: ${RPI}")
+		MESSAGE(STATUS "Setting _OPENGL_INCLUDE_PATH: ${RPI_GLES_PATH}/include")
+		MESSAGE(STATUS "Setting _OPENGL_LIB_PATH: ${RPI_GLES_PATH}/libs")                                                                                                         
+	SET(_OPENGL_INCLUDE_PATH ${RPI_GLES_PATH}/include)
```
Comment (lynxlynxlynx) [inline CMakeLists.txt:321]:
```suggestion
		SET(_OPENGL_INCLUDE_PATH ${RPI_GLES_PATH}/include)
```

**Pull request merged**